pkg/sync/sync.go

```diff
@@ -894,7 +894,7 @@ func matchTwoStar(p string, s []string) bool {
 	if !ok {
 		return false
 	}
-	for i := 1; i <= len(s); i++ {
```

Comment:
please add the failed case

Reply:
<img width="1744" alt="image" src="https://github.com/juicedata/juicefs/assets/31313340/6b19cd25-ee55-47ea-b74f-4a688dbf315c">